### PR TITLE
[Datafactory] Adding new API to get GitHub access token

### PR DESCRIPTION
**Pull request by hvermis**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json [hunk 1]
```diff
@@ -3177,6 +3228,36 @@
...
+      "required": [
+        "gitHubAccessCode",
+        "gitHubAccessTokenBaseUrl"
+      ]
+    },
+    "GitHubAccessTokenResponse": {
+      "description": "Get GitHub access token response definition.",
+      "properties": {
+        "gitHubAccessToken": {
+          "description": "GitHub access token.",
```
Comment (hovsepm) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json]: `"readOnly": true` ?

**Comment by hovsepm** [comment]
```
error: 
operationId: Factories_GetGitHubAccessToken
scenario: Factories_GetGitHubAccessToken
source: response
responseCode: '200'
severity: 0
errorCode: OBJECT_ADDITIONAL_PROPERTIES
errorDetails:
  code: OBJECT_ADDITIONAL_PROPERTIES
  params:
    - - tags
      - location
      - eTag
      - properties
      - type
      - name
      - id
  message: >-
    Additional properties not allowed:
    tags,location,eTag,properties,type,name,id
  path: ''
  title: /definitions/GitHubAccessTokenResponse
  description: Get GitHub access token response definition.
  position:
    line: 3252
    column: 34
  url: >-
    specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json
  jsonPosition: !<tag:yaml.org,2002:js/undefined> ''
  jsonUrl: >-
    specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/examples/Factories_GetGitHubAccessToken.json
```

Seems like example does not return the type that is described in the swagger. Please fix.

**Comment by hovsepm** [comment]
@hvermis there are bunch of existing Model Validator Errors not related to your PR. Please fix them in the future ;)

**Review by hovsepm — CHANGES_REQUESTED** [review]
waiting for example response message fix.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json [hunk 2]
```diff
@@ -395,6 +395,57 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DataFactory/factories/{factoryName}/getGitHubAccessToken": {
```
Comment (simongdavies) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json]: Do you expect this action to be used in an ARM template? If so you may want to change the name so it begins with list, if you do this the action will be able to be invoked from a template , you also need to make changes to your manifest to enable this functionality as there is data in the POST body.

Comment (hvermis) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json]: No, this will not be used in ARM templates.

**Comment by hvermis** [comment]
Will address them in next PR

---
In reply to: [419502847](https://github.com/Azure/azure-rest-api-specs/pull/3827#issuecomment-419502847) [](ancestors = 419502847)

**Review by hovsepm — APPROVED** [review]

**Pull request merged**